src/fs.bash

```diff
@@ -30,9 +30,27 @@ fs.is_symlink() {
     return 1
 }
 
-# check whether file is a symlink
+# check whether file is symlinked to a package
 fs.is_ellipsis_symlink() {
-    if [[ -L "$1" && "$(readlink "$1")" == $ELLIPSIS_PATH/* ]]; then
+    fs.is_target_symlink "$1" "$ELLIPSIS_PACKAGES"
```

Comment:
I think it's better to keep the original behavior here, to keep things backwards compatible.

`fs.is_target_symlink "$1" "$ELLIPSIS_PATH"` or even better, just reference the new `fs.is_ellipsis_core_symlink` (which should exactly mimic the old behavior if I'm not mistaken).

We can then gradually replace the old `fs.is_ellipsis_symlink` with `fs.is_package_symlink` where apropriate.

Reply:
On my side, I consider this as a bug, and this became an issue when I installed my ellipsis packages in a different root: 
```
ELLIPSIS_PACKAGES=/home/user.local/pkgs_ellipsis
ELLIPSIS_PATH=/home/user/.local/opt/ellipsis
```

In this context, `fs.is_ellipsis_symlink` was always returning `1` in my `ellipsis.sh` packages. And it is basically useless to use this function in my context. The current implementation check if a link belong to ellipsis instead of checking if a link belong to an ellipsis package, which IMO does not serve any purpose, as you are not supposed to link any files from ellipsis.

The only valid exception I see is for `bin/ellipsis` that you may want to symlink in your PATH (and this is only relevant if don't want to patch your `.bashrc` to run ellipsis init hooks). What does matter here is related to the package you installed.

My initial approach was to keep the initial feature available, and avoiding to patch all existing packages that use the `fs.is_ellipsis_symlink`, so I reimplemented the same function but with a different name: `fs.is_ellipsis_core_symlink`. 

TBH, I don't really like that, I would have prefered something like this, which seems to be cleaner and more easily understandable from the user perspective:
```
fs.is_package_symlink
fs.is_ellipsis_symlink
```
But it has the unfortunate consequence to not fix all existing package that already use `fs.is_ellipsis_symlink`. So the question here is: do we want to fix this at the ellipsis level, or do we want to patch all ellipsis packages to fit a to a cleaner API? This one is on your side :D

Reply:
I have another proposal, and I think it makes sense;

```
fs.is_ellipisis_symlink() {
  fs.is_ellipsis_core_symlink || fs.is_ellipsis_package_symlink
}
```
This way the original bug is fixed, and core and package functions are actually the specific versions for the now fixed is_ellipsis_symlink function which does check both as the name is already suggesting.